src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -206,6 +209,26 @@ export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) =>
     });
   }, [props.core]);
 
+  useEffect(() => {
+    setIsSummaryAgentAvailable(false);
+    const fetchSummaryAgent = async () => {
+      if (selectedDataset.current?.dataSource?.id) {
+        try {
```

Comment:
we should support datasource not enabled case

Reply:
Thanks for the comments, updated

Reply:
We may not need to have `if (selectedDataset.current?.dataSource?.id)`, call `await checkAgentsExist()` with `selectedDataset.current?.dataSource?.id` directly should be good. because:
1. when data source is not enabled, the selectedDataset.dataSource is undefined, in such case, we passed `undefined` to `checkAgentsExist`, this will make it to check local cluster
2. when data source is enabled, but the selected dataset is a local cluster, the selectedDataset.dataSource is undefined as well, in this case, we passed `undefined` to `checkAgentsExist` which makes it to call local cluster.